Implement support for the `drop-shadow` filter

EnnuiL opened the pull request:
Huh, I'm genuinely surprised that nobody had implemented it so far;

This PR implements the WebRender conversions that are necessary in order to make `drop-shadow` work; However, with the goal of making `currentColor` work, the code is uh, not really as nice as I wish, since filter conversions ended needing to be split into its own trait and its own method that accepts a current color; I assume that there is a nicer way of doing this, and yeah, pointers to the right direction would be appreciated

The results, however, are pretty nice though!
![Cropped screenshot of Cookie Clicker showing the drop shadows working on Servo](https://github.com/servo/servo/assets/85590273/7c415982-bec3-4462-b2d9-d67ae6c11970)

About test results, `css/css-color/currentcolor-003.html` got a bit closer to passing (although `currentColor` handling still needs improvements) and I assume `css/filter-effects/filters-drop-shadow-003.html` still fails due to positioning issues

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #5187

<!-- Either: -->
- [x] There are tests for these changes OR
- [ ] These changes do not require tests because ___

EnnuiL committed:
Implement support for `drop-shadow`

EnnuiL committed:
Clean up remnant from early attempts

mrobinson commented:
Nicely done. This is modifying files in `style` which are normally done upstream in Gecko, so perhaps @Loirooriol who is the resident expert on the style crate can take a look here.

EnnuiL committed:
Fix misleading comments on GenericSimpleShadow
If Servo-specific `style` changes will need to be upstreamed anyway, I might as well fix a thing that had thrown me off!

EnnuiL commented on components/style/values/generics/effects.rs:
```diff
@@ -110,9 +110,9 @@ pub use self::GenericFilter as Filter;
+    /// Horizontal offset.
     pub horizontal: SizeLength,
-    /// Vertical radius.
+    /// Vertical offset.
```
Ah, fair enough

EnnuiL committed:
Revert "Fix misleading comments on GenericSimpleShadow"

This reverts commit cdc810b826ac082041adc212c24649ee3b86ca0a.

Loirooriol reviewed (approved):
Looks reasonable, thanks!

EnnuiL committed:
Clean up an import

EnnuiL commented:
I'm so confused; when I saw the CI error, I thought "welp, I forgot to run this instance of a test with `drop-shadow`s and now the test's complaining about a false-positive being corrected", but uhhh, attempting to run the `css-masking` tests locally has instead opened the floodgate for 96 failures that includes way more than what CI complains about, with that huge number barely changing once attempting to revert the changes

So, yeah, this PR will need to update its test expectations in order to account for these corrected false-positives, but yeah, I guess my machine simply hates the concept of CSS Masking; Maybe I could attempt to manually add those?

EnnuiL committed:
Update test expectations

jdm commented:
@bors-servo try

EnnuiL committed:
Fix missing expectation on Layout 2013

EnnuiL commented:
...welp, I forgot about the Legacy Layout; all test expectations should have been fixed now
